module/VuFind/tests/unit-tests/src/VuFindTest/Solr/WriterTest.php

```diff
@@ -145,63 +120,64 @@ public function testDeleteAll()
-        };
-        $connector->expects($this->once())->method('write')->with($this->callback($callback));
-        $ct = $this->getMockChangeTracker();
-        $ct->expects($this->exactly(2))->method('markDeleted')
-            ->withConsecutive(['biblio', 'foo'], ['biblio', 'bar']);
-        $writer = new Writer($this->getSearchService($bm), $ct);
-        $writer->deleteRecords('Solr', ['foo', 'bar']);
+        $deleteDoc = new DeleteDocument();
+        $deleteDoc->addKeys(['foo', 'bar']);
+        $command = new WriteDocumentCommand('Solr', $deleteDoc);
+        $this->getWriter($command, ['core' => 'biblio'])
+            ->deleteRecords('Solr', ['foo', 'bar']);
     }
 
     /**
-     * Get mock backend manager
+     * Get mock change tracker
      *
-     * @return BackendManager
+     * @return ChangeTracker
      */
-    protected function getBackendManagerWithMockSolr()
+    protected function getMockChangeTracker()
     {
-        $sm = new \Laminas\ServiceManager\ServiceManager();
-        $pm = new BackendManager($sm);
-        $mockBackend
-            = $this->getMockBuilder(\VuFindSearch\Backend\Solr\Backend::class)
-                ->disableOriginalConstructor()
-                ->onlyMethods(['getConnector', 'getIdentifier'])
-                ->getMock();
-        $handlerMap = new HandlerMap();
-        $mockConnector
-            = $this->getMockBuilder(\VuFindSearch\Backend\Solr\Connector::class)
-                ->setConstructorArgs(
-                    [
-                        'http://localhost:8983/solr/biblio',
-                        $handlerMap,
-                        null
-                    ]
-                )
-                ->onlyMethods(['write'])
-                ->getMock();
-        $mockBackend->expects($this->any())->method('getConnector')
-            ->will($this->returnValue($mockConnector));
-        $mockBackend->expects($this->any())->method('getIdentifier')
-            ->will($this->returnValue('Solr'));
-        $sm->setService('Solr', $mockBackend);
-        return $pm;
+        return $this->getMockBuilder(\VuFind\Db\Table\ChangeTracker::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['markDeleted'])
+            ->getMock();
     }
 
     /**
-     * Get mock change tracker
+     * Create a mock search service for a single command and its result
      *
-     * @return ChangeTracker
+     * @param object $expectedCommand Expected command class
+     * @param mixed  $result          Result to return for the invoked command
+     *
+     * @return MockObject&SearchService
      */
-    protected function getMockChangeTracker()
+    protected function getMockSearchService($expectedCommand, $result)
     {
-        return $this->getMockBuilder(\VuFind\Db\Table\ChangeTracker::class)
+        $resultCommand = $this->getMockBuilder(WriteDocumentCommand::class)
```

Comment:
Would it make sense for this to be `get_class($expectedCommand)` rather than a hard-coded value, or am I misunderstanding something?

Reply:
Makes sense, done. They all were instances of `WriteDocumentCommand`, but `get_class` makes the intention more obvious.